### ISSUE TITLE
[AI] Improve AI performance by 5-10x

### ISSUE THREAD

**Issue by cpojer** [nkzw-tech/athena-crisis#15]
The Athena Crisis AI works by executing all actions for a turn and returning them to the user. It's critical for the AI to be fast. The goal of this task is to speed up the AI by 5-10x while maintaining its difficulty level. Ideally it has the same or stronger behavior, while being faster. All tests in this repo need to pass with the improved AI, however if a snapshot test changes while retaining behavior, that is acceptable.

### AI Info
The current AI is called [`DionysusAlpha`](https://github.com/nkzw-tech/athena-crisis/blob/main/dionysus/DionysusAlpha.tsx) and its code can be found there. There are also many AI specific helpers in [`dionysus/lib`](https://github.com/nkzw-tech/athena-crisis/tree/main/dionysus/lib). Check out the [AI Playground](http://athenacrisis.com/open-source/playground/ai).

### Notes
* **Choice of Tooling:** The AI has to be written in JavaScript and be maintainable. This is not a code-golf competition or challenge about porting code to another programming language.
* **Radius Performance:** Improvements made as part of https://github.com/nkzw-tech/athena-crisis/issues/9 will count towards that issue. If we find that there is a new performance lower-bound based on improvements to that issue, we may adjust the target of this issue down to make sure the goal is achievable.

**Comment by mohamedsalem331** [comment]
@cpojer is the dionysus directory contains all logic needed to understand current ai, iam new to the repo. and hoping i can contribute.

**Comment by cpojer** [comment]
Yes! Check out the AI and prior sections in the docs: https://athenacrisis.com/open-source/core-concepts/ai

**Comment by cpojer** [comment]
AI perf is good now.
